src/components/ScenarioCard.test.tsx

```diff
@@ -414,4 +414,83 @@ describe("tooltip functionality", () => {
       expect.stringContaining("cursor-help"),
     );
   });
+
+  describe("ScenarioCard robustness with non-string values", () => {
+    const baseScenario: Scenario = {
+      id: "robust-1",
+      name: "Scenario A",
+      description: "Desc",
+      pathwayType: "Policy",
+      modelYearEnd: "2030",
+      modelTempIncrease: 1.5,
+      regions: ["EU", "US"],
+      sectors: [{ name: "Power" }],
+      publisher: "RMI",
+      publicationYear: "2024",
+      overview: "x",
+      expertRecommendation: "x",
+      dataSource: {
+        description: "x",
+        url: "https://example.com",
+        downloadAvailable: false,
+      },
+    };
+
+    const renderWithRouter = (scenario: Scenario, searchTerm = "") =>
+      render(
+        <MemoryRouter>
+          <ScenarioCard
+            scenario={scenario}
+            searchTerm={searchTerm}
+          />
+        </MemoryRouter>,
+      );
+
+    it("does not crash when highlighting numeric fields", () => {
+      const s: Scenario = {
+        ...baseScenario,
+        // Using a double cast to satisfy TS, but this still presents as numeric at runtime.
+        modelYearEnd: 2030 as unknown as string, // number on purpose
+        publicationYear: 2024 as unknown as string, // number
```

Comment:
@jdhoffa @cjyetman Just an FYI on a weird workaround for linting, in case you need it, this double cast (`as unknown as string`) replaces an `as any` cast which ESLint got cranky about. 

See https://github.com/RMI/pbtar/pull/268/commits/46242adac338c99e8972f273f6ddba9546ec4141